eng/DownloadSharedSource.ps1

```diff
@@ -2,29 +2,89 @@
 $ErrorActionPreference = 'Stop'
 $ProgressPreference = 'SilentlyContinue'
 
-function DownloadAll([string[]]$files, [string]$baseUrl, [string]$downloadPath)
-{
-    foreach ($file in $files)
-    {
-        Write-Host "Downloading" $file
-        $text = (Invoke-WebRequest -Uri "$baseUrl/$file").Content
-        $text.Trim() | Out-File (Join-Path $downloadPath $file)
+$sparseCheckoutFile = ".git/info/sparse-checkout"
```

Comment:
I know you copied this from another script but I think you can simplify this to something like:

```
$repoRoot = "$PSScriptRoot/.."
$clonedPath = "$repoRoot/../core-shared-source"
$azCoreSharedPath = "sdk/core/Azure.Core/src/Shared/"
$armCoreSharedPath = "sdk/resourcemanager/Azure.ResourceManager/src/Shared/"

Write-Host "git clone --no-checkout --filter=tree:0 https://github.com/Azure/azure-sdk-for-net.git $clonedPath"
git clone --no-checkout --filter=tree:0 https://github.com/Azure/azure-sdk-for-net.git $clonedPath
Push-Location $clonedPath
Write-Host "git sparse-checkout init"
git sparse-checkout init
Write-Host "git sparse-checkout set --no-cone $azCoreSharedPath $armCoreSharedPath"
git sparse-checkout set --no-cone $azCoreSharedPath $armCoreSharedPath
Write-Host "git checkout"
git checkout
Pop-Location

$files = @('AsyncLockWithValue.cs', 'ClientDiagnostics.cs', 'DiagnosticScope.cs', 'DiagnosticScopeFactory.cs', 'ContentTypeUtilities.cs', 'HttpMessageSanitizer.cs',
    'OperationInternalBase.cs', 'OperationInternal.cs', 'OperationInternalOfT.cs', 'TaskExtensions.cs', 'Argument.cs', 'Multipart/MultipartFormDataContent.cs',
    'Multipart/MultipartContent.cs', 'AzureKeyCredentialPolicy.cs', 'AppContextSwitchHelper.cs',
    'OperationPoller.cs', 'FixedDelayWithNoJitterStrategy.cs', 'SequentialDelayStrategy.cs',
    'ForwardsClientCallsAttribute.cs', 'AsyncLockWithValue.cs', 'VoidValue.cs')
foreach ($file in $files)
{
   $clonedFilePath = Join-Path $clonedPath $azCoreSharedPath $file
   $assetsFilePath = Join-Path $repoRoot "src/assets/Azure.Core.Shared"
   Copy-Item $clonedFilePath $assetsFilePath 
}

$files = 'SharedExtensions.cs', 'ManagedServiceIdentityTypeV3Converter.cs'
foreach ($file in $files)
{
   $clonedFilePath = Join-Path $clonedPath $armCoreSharedPath $file
   $assetsFilePath = Join-Path $repoRoot "src/assets/Management.Shared"
   Copy-Item $clonedFilePath $assetsFilePath 
}
```

Reply:
I'm fine with putting it under a gitignored directory as well.